Cython/Compiler/Symtab.py

```diff
@@ -922,20 +923,44 @@ def find_imported_module(self, path, pos):
     def lookup(self, name):
         # Look up name in this scope or an enclosing one.
         # Return None if not found.
-        name = self.mangle_class_private_name(name)
-        return (self.lookup_here(name)
-            or (self.outer_scope and self.outer_scope.lookup(name))
-            or None)
+
+        mangled_name = self.mangle_class_private_name(name)
+        entry = (self.lookup_here(name)  # lookup here also does mangling
+                or (self.outer_scope and self.outer_scope.lookup(mangled_name))
+                or None)
+        if entry:
+            return entry
+
+        # look up the original name in the outer scope
+        # Not strictly Python behaviour but see https://github.com/cython/cython/issues/3544
+        entry = (self.outer_scope and self.outer_scope.lookup(name)) or None
+        if entry and entry.is_pyglobal:
+            self._lookup_python_mangled_warning(entry.pos, name)
+        return entry
 
     def lookup_here(self, name):
         # Look up in this scope only, return None if not found.
+
+        # if the name is already declared in the scope then return it, even if
+        # mangling should give us something else. This is to support things like
+        # global __foo which makes a declaration for __foo
```

Comment:
Hmm, is it possible that a scope declares both a mangled and an unmangled double underscore name? Like a `cdef extern` C name and a Python variable with the same name, only the latter of which gets mangled now?

Reply:
Yes - this looks the wrong way round. Fixed now.